hie-base/Haskell/Ide/Engine/SemanticTypes.hs

```diff
@@ -82,16 +89,13 @@ instance ValidResponse TypeInfo where
 
 instance ToJSON TypeResult where
   toJSON (TypeResult s e t) =
-      object [ "start" .= posToJSON s
-             , "end" .= posToJSON e
-             , "type" .= t
+      object [ "start" .= toJSON s
+             , "end"   .= toJSON e
+             , "type"  .= t
              ]
 
 instance FromJSON TypeResult where
-  parseJSON (Object v) = TypeResult
-    <$> (jsonToPos =<< (v .: "start"))
-    <*> (jsonToPos =<< (v .: "end"))
-    <*> v .: "type"
+  parseJSON (Object v) = TypeResult <$> v .: "start" <*> v .: "end" <*> v .: "type"
```

Comment:
Ok, added withObject wherever I could. Hope I got them all. Learned something new in the process.